decent_ci_run.rb

```diff
@@ -72,12 +72,23 @@
   puts "Successfully cloned decent_ci repository."
 
   FileUtils.cd(run_dir)
+  
+  if !config["purge_build_dirs"].nil? and config["purge_build_dirs"] then
+    puts "Purging Build Dirs in #{run_dir}"
+    Dir.entries(run_dir).each { |entry|
+      if File.directory?(entry) and entry =~ /^.+-[a-f0-9]+-.+-.+$/ then
+        puts "Purging matched directory: #{entry}"
+        FileUtils.remove_entry_secure(entry, true)
```

Comment:
Is it worth having a specific rescue around this line to trap for errors specifically related to removing the folder?  Or could you put a 5 second timeout after the call and then check to see if the folder still exists and raise an exception if it does?  I dunno, just thinking outloud about how to best deduce the problem.

Reply:
Hmm, it probably does make sense to have a specific rescue around it. But I think just for a logging, I don't think we want to do retry logic here since it will simply retry on the next run around (at which point the suspect files will have been long since stopped being used for the compilation)